src/components/PaginationButton.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <button
+    :class="[
+      'pagination-button',
+      ...($slots.default ? [`pagination-button--${direction}`] : []),
+      {'pagination-button--disabled': disabled}
+    ]"
+    @click="!disabled && $emit('click')">
+    <app-icon
+      v-if="direction === 'left'"
+      :size="16"
+      name="caret-left"/>
+    <slot/>
+
+    <app-icon
+      v-if="direction === 'right'"
+      :size="16"
+      name="caret-right"/>
+  </button>
+</template>
+
+<script setup>
+import AppIcon from '@/components/AppIcon'
+
+defineProps({
+  direction: {
+    type: String,
+    default: 'left',
```

Comment:
I am not sure if the default should be left.
1. I don't thing it's a logical conclusion
2. we are not using it implicitly in any declaration
3. you could use it in the conditional class check (not checking slots)